src/TextureManager.cpp

```diff
@@ -6,17 +6,32 @@
 
 #include "GameInternal.h"
 
-SDL_Texture* TextureManager::loadTexture(const char* fileName)
-{
-	auto it = this->texture_cache.find(fileName);
-	if (it != this->texture_cache.end()) {
-		return it->second;
+
+void TextureManager::addSingleTexture(TexturesEnum texture, const char* filePath) {
```

Comment:
This is problematics as it doesn't allow adding automatically provided textures (i.e. map tilesheets) - which would require a generic addTexture method without an identifier